prow/cmd/pipeline/controller.go

```diff
@@ -586,31 +591,72 @@ func makePipelineGitResource(pj prowjobv1.ProwJob) *pipelinev1alpha1.PipelineRes
 	return &pr
 }
 
-// makePipeline creates a PipelineRun from a prow job using the PipelineRunSpec defined in the prow job
-func makePipelineRun(pj prowjobv1.ProwJob, pr *pipelinev1alpha1.PipelineResource) (*pipelinev1alpha1.PipelineRun, error) {
+// makePipeline creates a PipelineRun and a slice of PipelineResources from a ProwJob using the
+// PipelineRunSpec and git refs.
+func makeResources(pj prowjobv1.ProwJob) (*pipelinev1alpha1.PipelineRun, []pipelinev1alpha1.PipelineResource, error) {
+	// First validate.
 	if pj.Spec.PipelineRunSpec == nil {
-		return nil, errors.New("no PipelineSpec defined")
+		return nil, nil, errors.New("no PipelineSpec defined")
 	}
+	buildID := pj.Status.BuildID
+	if buildID == "" {
+		return nil, nil, errors.New("empty BuildID in status")
+	}
+	if err := config.ValidatePipelineRunSpec(pj.Spec.Type, pj.Spec.ExtraRefs, pj.Spec.PipelineRunSpec); err != nil {
+		return nil, nil, fmt.Errorf("invalid pipeline_run_spec: %v", err)
+	}
+
 	p := pipelinev1alpha1.PipelineRun{
-		ObjectMeta: pipelineMeta(pj),
+		ObjectMeta: pipelineMeta(pj.Name, pj),
 		Spec:       *pj.Spec.PipelineRunSpec.DeepCopy(),
 	}
-	buildID := pj.Status.BuildID
-	if buildID == "" {
-		return nil, errors.New("empty BuildID in status")
+
+	// Add parameters instead of env vars.
+	env, err := downwardapi.EnvForSpec(downwardapi.NewJobSpec(pj.Spec, buildID, pj.Name))
+	if err != nil {
+		return nil, nil, err
+	}
+	for _, key := range sets.StringKeySet(env).List() {
+		val := env[key]
+		// TODO: make this handle existing values/substitutions.
```

Comment:
@cjwagner what do you mean by that ?

Reply:
As is this just appends the Prow parameters to the list without regard for existing parameters. If a Prow parameter is explicitly specified by the user we probably will want to error (at config validation time as well) or replace the value. I'm not sure which option is more appropriate. WDYT?